### PR TITLE
removes the syntax highlighting dependency on Net

### PR DESCRIPTION
**Pull request by finanalyst**
@coke @dontlaugh This new plugin uses @patrickbkr 's very new entirely Raku syntax highlighter (Rainbow).
1. The plugin removes the dependency on the atom-highlighter, which was written originally in coffeescript. 
2. In turn, the removal of atom-highlighter removes entirely the dependency on npm and a large number of other javascript dependency in a webpack. This is a major simplification for anyone wanting to install doc-website, but who does not have npm (or equivalent).
3. Rainbow offers more categories for coloration than the atom-highlighter:
  - instead of variable, we now have scalar, array & hash
  - three regex categories, where none existed before
  - two pod (rakudoc) categories
  - several delimiters
  For the time being I have not differentiated between the variables and delimiters, and kept the colours used by atom-highlighter.
4. To change the colours, there is a list of variables in rainbow-dark.scss and rainbow-dark.scss ($code-xxx-color). Change the color definition. These are then interpolated into the appropriate classes.

Note that if the new highlighter does not work, then the previous highlighter can be re-enabled by swapping 'highliter' for 'rainbow' in the plugin config file.

### CODE REVIEW

**Comment by finanalyst** [comment]
The effect of the new highlighter will be seen in https://docs-dev.raku.org [eg operators](https://docs-dev.raku.org/language/operators#Operator_classification)

**Comment by dontlaugh** [comment]
@finanalyst Do any of the commands run in CI need updating/removing? https://github.com/Raku/doc-website/blob/main/.buildkite/pipeline.yaml#L6

**Comment by finanalyst** [comment]
@dontlaugh I dont understand why build kite is failing.
- The change is to replace one plugin with another.
- The new plugin does not need any of the Net infrastructure used by atom-highlighter
- So, I thought that I would disable one the one that uses atom-highlighter, but leave the infrastructure in place.
- I cant see why removing the infrastructure would stop this automatic stop.

I'll add the changes now to remove the Net infrastructure, and see what happens

**Comment by finanalyst** [comment]
@dontlaugh Should this branch be called 'docs-dev'? I called it 'dev'. Will the rendered files be containered into docs-dev.raku.org?

**Comment by finanalyst** [comment]
@dontlaugh I tried to cancel an intermediate build but could not work out how to do this. Build-kite documentation indicates there is a setting that automatically cancels intermediate builds if two pushes are close together.
Is there a reason we cant use this setting?

**Comment by finanalyst** [comment]
@dontlaugh @coke The new plugin seems to be working VERY much slower than atom-perl6-highlighter. So I think !!!?? this will need some investigation

**Comment by patrickbkr** [comment]
I've noticed that the highlighter is slow as well. It's probably not an issue with your setup, but with Rainbow itself. I do need to look into this at some point, it just didn't make it to the top of my priority list. @finanalyst Is this a blocker for you? If so, I can bump it to the top.

**Comment by dontlaugh** [comment]
The build is timing out, that's why it's failing. No container will be built until it's green.

Also note that "dev" nor "docs-dev" are automatically deploying right now. That's my bad: I needed to fail prod over to dev a couple of weeks ago.

Let's work on getting this tested locally, then getting the build green. I'll upgrade the build environment tonight. Note that this build is not running the latest Rakudo.

**Comment by finanalyst** [comment]
@patrickbkr Thanks for responding here. I was just going to raise an issue on the Rainbow repo.
Doc-website uses Sam McVey's atom-perl6-highlighter at present, but it is the only non-Raku  element of the tool-chain to the rendered files. The problem is that we need a whole .Net infrastructure to get it to run. 
So, its not a blocker per se, but it would be much more elegant to be able to use Rainbow.
But we can't use Rainbow until its much faster.

File: META6.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "Raku-Official-Documentation",
   "description": "Tool chain for creating the Raku official documentation site",
-  "version": "0.1.1",
+  "version": "0.1.2",
```
[the remark this answers is not in the document]
Comment (finanalyst) [reply META6.json:4]: @dontlaugh more abstractly: Rainbow could be merged into MAIN so long as `configs/02-plugins.raku` is not changed. 
The `atom-perl6-highlighter` is used by the `hiliter` plugin, while `Rainbow` is used by the `rainbow` plugin. They are 1-1 interchangeable as far as `Collection` is concerned.

**Comment by dontlaugh** [comment]
I see the build is green now. Push to **docs-dev** to build a pre-prod container

https://buildkite.com/raku/doc-website/builds/6999

I'll make it so that docs-dev shows up on https://docs-dev.raku.org/ again

**Comment by finanalyst** [comment]
@dontlaugh Thanks for pushing to docs-dev.

FYI Rainbow v0.2 is 38x faster than Rainbow v0.1, and about 5x faster than atom-perl6-highlighter (the nodejs version).

@coke The colour palette  of the syntax is the same as atom-perl6-highlighter, but the categories are different. Also, I have used the same colours for multiple categories. It would be good for someone to revise the coloration.

I was thinking we could ask for some more eyes on the coloration before committing the changes to main.

**Review by dontlaugh — APPROVED** [review]
I'm approving this, since I think it can be merged at the discretion of @finanalyst ; Even if Rainbow introduces some regressions, I'm inclined to forge ahead. More eyes will find the bugs faster.

Note that the Rainbow plugin's bug tracker is hosted on Sourcehut. https://todo.sr.ht/~patrickb/Rainbow

**Comment by dontlaugh** [comment]
@finanalyst yes, let's test that configuration

**Comment by finanalyst** [comment]
Just realised, I removed epub generation! Re-enabling.

**Comment by dontlaugh** [comment]
I've pushed this branch to docs-dev https://buildkite.com/raku/doc-website/builds/7024

**Comment by dontlaugh** [comment]
Any blocker on this?

**Comment by dontlaugh** [comment]
#419 is where comments on the new code highlighting style can be collected

**Pull request merged**